scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -758,7 +762,8 @@ object FormatWriter {
     if (pipe == '|') leadingPipeSpace else compileStripMarginPattern(pipe)
   @inline
   private def compileStripMarginPattern(pipe: Char) =
-    Pattern.compile(s"(?<=\\n)\\h*(?=[$pipe])")
+    // capture group is used to pass pipe character to distant replaceAll.
+    Pattern.compile(s"\n\\h*(\\${pipe})", Pattern.MULTILINE)
```

Comment:
What if pipe contains a closing parenthesis? I admit that my solution wouldn't work for pipe containing a closing bracket but I'm not sure why this approach is fully insulated.

Reply:
Challenge-response helps build better code.  You make a good (friendly argument) for
some explicit test cases. 

By my reading & intent, a closing parenthesis of any ilk: round, curly, square.
It should be escaped by the backslash (which, for future readers,
needs to be doubled because it is in an interpolated string). 

Once the pipe character is in the capture group, it should able to be used by
replaceAll with no problems.  Based on no data, that is my story..

Does it parse that way for you?

It is late in my day today and I have promises to keep but I'll take a run at stripMargin tests
which check out the usual suspects.

Reply:
Fixed by adding three unit tests to StripMargin.stat for alternate pipe characters, including
Right or closing round parenthesis.